mlir/unittests/Analysis/Presburger/CoalesceTest.cpp

```diff
@@ -0,0 +1,337 @@
+  cutConstraint1.push_back(0);
+  cutConstraint1.push_back(2);
+  SmallVector<ArrayRef<int64_t>, 8> cutting = {cutConstraint1};
+  expectContainedFacet(true, ineq, bs, cutting);
+}
+
+TEST(CoalesceTest, containedFacet2) {
+  PresburgerSet set =
+      setFromString("(x, y) : (x >= 0 and x <= 3 and y >= 0 and y <= 3)");
+  FlatAffineConstraints bs = set.getFlatAffineConstraints()[0];
+  SmallVector<int64_t, 8> ineq;
+  ineq.push_back(1);
+  ineq.push_back(0);
+  ineq.push_back(0);
+  SmallVector<int64_t, 8> cutConstraint1;
+  cutConstraint1.push_back(1);
+  cutConstraint1.push_back(0);
+  cutConstraint1.push_back(2);
+  SmallVector<int64_t, 8> cutConstraint2;
+  cutConstraint2.push_back(1);
+  cutConstraint2.push_back(0);
+  cutConstraint2.push_back(1);
+  SmallVector<ArrayRef<int64_t>, 8> cutting = {cutConstraint1, cutConstraint2};
+  expectContainedFacet(true, ineq, bs, cutting);
+}
+
+TEST(CoalesceTest, containedFacet3) {
+  PresburgerSet set =
+      setFromString("(x, y) : (x >= 0 and x <= 3 and y >= 0 and y <= 3)");
+  FlatAffineConstraints bs = set.getFlatAffineConstraints()[0];
+  SmallVector<int64_t, 8> ineq;
+  ineq.push_back(1);
+  ineq.push_back(0);
+  ineq.push_back(0);
+  SmallVector<int64_t, 8> cutConstraint1;
+  cutConstraint1.push_back(1);
+  cutConstraint1.push_back(0);
+  cutConstraint1.push_back(-5);
+  SmallVector<ArrayRef<int64_t>, 8> cutting = {cutConstraint1};
+  expectContainedFacet(false, ineq, bs, cutting);
+}
+
+TEST(CoalesceTest, containedFacet4) {
+  PresburgerSet set =
+      setFromString("(x, y) : (x >= 0 and x <= 3 and y >= 0 and y <= 3)");
+  FlatAffineConstraints bs = set.getFlatAffineConstraints()[0];
+  SmallVector<int64_t, 8> ineq;
+  ineq.push_back(1);
+  ineq.push_back(0);
+  ineq.push_back(0);
+  SmallVector<int64_t, 8> cutConstraint1;
+  cutConstraint1.push_back(0);
+  cutConstraint1.push_back(1);
+  cutConstraint1.push_back(2);
+  SmallVector<int64_t, 8> cutConstraint2;
+  cutConstraint2.push_back(1);
+  cutConstraint2.push_back(0);
+  cutConstraint2.push_back(-5);
+  SmallVector<ArrayRef<int64_t>, 8> cutting = {cutConstraint1, cutConstraint2};
+  expectContainedFacet(false, ineq, bs, cutting);
+}
+
+void expectWrapping(Optional<SmallVector<int64_t, 8>> expected,
+                    FlatAffineConstraints bs, SmallVector<int64_t, 8> valid,
+                    SmallVector<int64_t, 8> invalid) {
+  if (!expected) {
+    EXPECT_FALSE(wrapping(bs, valid, invalid).hasValue());
+  } else {
+    auto result = wrapping(bs, valid, invalid);
+    EXPECT_TRUE(result.hasValue());
+    if (!result) {
+      EXPECT_TRUE(false);
+    } else {
+      EXPECT_TRUE(sameConstraint(result.getValue(), expected.getValue()));
+    }
+  }
+}
+
+TEST(CoalesceTest, wrapping) {
+  PresburgerSet set1 = setFromString("(x,y) : (x = y and y <= 4 and y >= 0)");
+  FlatAffineConstraints bs1 = set1.getFlatAffineConstraints()[0];
+  SmallVector<int64_t, 8> valid1, invalid1, expected1;
+  valid1.push_back(-1);
```

Comment:
You can construct vectors directly with initializer lists:
` SmallVector<int64_t, 8> valid1{1, 2, 3, 4}`.